### PR TITLE
refactor: swap `pl$Series` and `as_polars_series.default`

### PR DESCRIPTION
**Pull request by eitsupi**
Related to #1012.

`as_polars_series.default()` used to call `pl$Series()`, but will be changed so that `pl$Series()` calls `as_polars_series()`.
This will ensure that every object is properly converted to a Series.

Python's `polars.Series.__init__()` internally converts various objects into a Series through a large number of conditional branches. This update is very similar to that.

### CODE REVIEW

File: R/as_polars.R [hunk 1]
```diff
@@ -367,13 +369,15 @@ as_polars_series.POSIXlt = function(x, name = NULL, ...) {
 #' @rdname as_polars_series
 #' @export
 as_polars_series.data.frame = function(x, name = NULL, ...) {
-  pl$DataFrame(unclass(x))$to_struct(name = name)
+  as_polars_df(x)$to_struct(name = name)
```
Comment (eitsupi) [inline R/as_polars.R:372]: This change is not directly related, but is a small refactoring to make object conversions more appropriate.

File: R/as_polars.R [hunk 2]
```diff
@@ -367,13 +369,15 @@ as_polars_series.POSIXlt = function(x, name = NULL, ...) {
...
 }
 
 
 #' @rdname as_polars_series
 #' @export
-as_polars_series.vctrs_rcrd = as_polars_series.data.frame
+as_polars_series.vctrs_rcrd = function(x, name = NULL, ...) {
+  pl$select(unclass(x))$to_struct(name = name)
+}
```
Comment (eitsupi) [inline R/as_polars.R:380]: Ditto.

File: R/as_polars.R [hunk 3]
```diff
@@ -303,11 +303,12 @@ as_polars_lf.RPolarsLazyGroupBy = function(x, ...) {
 
 #' To polars Series
 #'
-#' [as_polars_series()] is a generic function that converts an R object to a
-#' polars Series. It is basically a wrapper for [pl$Series()][pl_Series].
-#' @param x Object to convert into a polars Series
-#' @param name A string to use as the name of the Series.
-#' If `NULL` (default), the name of `x` is used or an unnamed Series is created.
+#' [as_polars_series()] is a generic function that converts an R object to
+#' [a polars Series][Series_class].
```
Comment (etiennebacher) [inline R/as_polars.R:307]: Should we have a list of R objects that can be converted to Series?

**Review by etiennebacher — APPROVED** [review]

File: R/as_polars.R (hunk 3 above)
Comment (eitsupi) [reply R/as_polars.R:307]: Definitely needed... See #425.
The documentation for the `arrow` package is helpful.

Comment (etiennebacher) [reply R/as_polars.R:307]: Ok, we can address #425 in full in another PR then
